File: algorithms/java/sorting_algorithms/insertion_sort.java
package sorting_algorithms;
import java.util.ArrayList;

public class insertion_sort {
	public static void main(String args[]) {
		ArrayList unsortedList = new ArrayList();
		for(int i = 0 ; i < 10 ;i++) {
			unsortedList.add(createRondomNum(100));
		}
		System.out.println("The unsorted list : "+unsortedList );
		ArrayList result = insertion_sort(unsortedList);
		System.out.println("The sorted list : "+result);
	}
	
	private static ArrayList insertion_sort(ArrayList list) {
		for(int i = 1;i<list.size();i++) {
			int key = (int) list.get(i);
			int j = i-1;
			while( j >= 0 && key < (int) list.get(j)) {
				list.set(j+1, (int) list.get(j));
				j--;
			}
			list.set(j+1,key);
		}
		return list;
	}

	private static int createRondomNum(int i) {
		int number = (int) ( i*Math.random()) ;
		return number;
	}

}

File: algorithms/python/sorting_algorithms/insertion_sort.py
import  random
def insertion_sort(_list):
    for i in range(1, len(_list)):
        key = _list[i]
        j = i - 1
        while j >= 0 and key < _list[j]:
            _list[j + 1] = _list[j]
            j -= 1
        _list[j + 1] = key
    return _list
unsortedList = []
for i in range(10):
    unsortedList.append(random.randint(0,100))
print(f'The unsorted list: {unsortedList}')
result = insertion_sort(unsortedList)
print(f'The sorted list: {result}')